calour/experiment.py

```diff
@@ -74,6 +76,18 @@ def __init__(self, data, sample_metadata, feature_metadata=None,
         # flag if data array is sparse (True) or dense (False)
         self.sparse = sparse
 
+    @property
+    def sparse(self):
+        return self._sparse
```

Comment:
i think better to actually test the sparsity of matrix as opposed to storing in _sparse, as otherwise maybe we forget to update _sparse? and we can get rid of _sparse